cluster/addons.go

```diff
@@ -219,23 +219,25 @@ func (c *Cluster) doAddonDeploy(ctx context.Context, addonYaml, resourceName str
 	if err != nil {
 		return &addonError{fmt.Sprintf("Failed to generate addon execute job: %v", err), isCritical}
 	}
-	err = c.ApplySystemAddonExcuteJob(addonJob)
-	if err != nil {
+
+	if err = c.ApplySystemAddonExcuteJob(addonJob, addonUpdated); err != nil {
 		return &addonError{fmt.Sprintf("%v", err), isCritical}
 	}
 	return nil
 }
 
-func (c *Cluster) StoreAddonConfigMap(ctx context.Context, addonYaml string, addonName string) error {
+func (c *Cluster) StoreAddonConfigMap(ctx context.Context, addonYaml string, addonName string) (bool, error) {
 	log.Infof(ctx, "[addons] Saving addon ConfigMap to Kubernetes")
+	updated := false
 	kubeClient, err := k8s.NewClient(c.LocalKubeConfigPath, c.K8sWrapTransport)
 	if err != nil {
-		return err
+		return updated, err
 	}
 	timeout := make(chan bool, 1)
 	go func() {
 		for {
-			err := k8s.UpdateConfigMap(kubeClient, []byte(addonYaml), addonName)
+
```

Comment:
do we ever timeout? What if this loop runs forever?

Reply:
Yes, we timeout here https://github.com/rancher/rke/pull/680/files#diff-e072bfb1478c2c4b66ce16353b1b3027R254